### DISCUSSION TITLE
monorepo: group per Dockerfile

### DISCUSSION THREAD

**Discussion by universam1** [renovatebot/renovate#11948] [Request Help] [answered]
Hi,

I have monorepo with many Dockerfiles separated by folder structure, and my goal is to group branches/MR **per** Dockerfile.
Currently it **groups all Dockerfiles of one type** in one MR instead of **all dependencies within this single Dockerfile** - resulting into broken inter-dependency relations.

I have no idea how to manage that except to create dozens of group definitions manually.

Thanks a lot for help!

**Comment by rarkins** [comment, marked as answer]
Are you using group rules already, or are you referring to how the same dependency is updated in all Dockerfiles within one MR?

You will want to use `baseDir` or `parentDir` along with `additionalBranchPrefix`, but I can't answer exactly how without knowing more about your repo. You can probably read https://github.com/renovatebot/renovate/discussions/7993 and work out a solution on your own though

**Reply by universam1** [reply]
That combination of templating to create unique branches does the trick!

Thank you for that hint, and highlighting the important pieces!
